noipy/main.py

```diff
@@ -34,6 +32,67 @@
     r'(?:/?|[/?]\S+)$', re.IGNORECASE)
 
 
+def auth_credentials(user_token, password, provider_class):
+    if provider_class.auth_type == 'T':
+        user_arg = user_token or utils.read_input(
+            "Paste your auth token: ")
+        auth = authinfo.ApiAuth(usertoken=user_arg)
+    else:
+        user_arg = user_token or utils.read_input(
+            "Type your username: ")
+        pass_arg = password or getpass.getpass("Type your password: ")
+        auth = authinfo.ApiAuth(user_arg, pass_arg)
+
+    return auth
+
+
+def execute_ddns_update(update_ddns, ip, hostname, auth,
+                        updater_options, provider, provider_class,
+                        exec_result, process_message=None):
+    response_code = None
+    response_text = None
+    if update_ddns:
+        ip_address = ip if ip else utils.get_ip()
+        if not ip_address:
+            process_message = "Unable to get IP address. Check connection."
+            exec_result = EXECUTION_RESULT_NOK
+        elif ip_address == utils.get_dns_ip(hostname):
+            process_message = "No update required."
+        else:
+            updater = provider_class(auth, hostname, updater_options)
+            print("Updating hostname '%s' with IP address %s "
+                  "[provider: '%s']..."
+                  % (hostname, ip_address, provider))
+            response_code, response_text = updater.update_dns(ip_address)
+            process_message = updater.status_message
+
+    return {
+        'exec_result': exec_result,
+        'response_code': response_code,
+        'response_text': response_text,
+        'process_message': process_message,
+    }
+
+
+def process_generic(url, updater_options,
```

Comment:
Same issues of `execute_ddns_update` function ([line 49](https://github.com/pv8/noipy/pull/25/files#diff-0892748911e7cd8288b84eace1cc35a2R49)):
```sh
77-79: Function `process_generic` has 5 arguments (exceeds 4 allowed). Consider refactoring. [structure]
```